pkg/calculator/handler.go

```diff
@@ -127,24 +127,22 @@ func (ec *EmissionCalculator) Apply(event bus.Event) {
 
 	// calculate and set the operational emissions for each
 	// metric type (CPU, Memory, Storage, and networking)
-	metrics := eventInstance.Metrics()
+	metrics := eventInstance.Metrics
 	for _, v := range metrics {
 		params.metric = &v
 		opEm, err := operationalEmissions(log.WithContext(context.Background(), ec.logger), interval, &params)
 		if err != nil {
-			ec.logger.Error("failed calculating operational emissions", "type", v.Name(), "error", err)
+			ec.logger.Error("failed calculating operational emissions", "type", v.Name, "error", err)
 			continue
 		}
-		params.metric.SetEmissions(v1.NewResourceEmission(opEm, v1.GCO2eqkWh))
+		params.metric.Emissions = v1.NewResourceEmission(opEm, v1.GCO2eqkWh)
 		// update the instance metrics
 		metrics.Upsert(params.metric)
 	}
 
-	eventInstance.SetEmbodiedEmissions(
-		v1.NewResourceEmission(
-			embodiedEmissions(interval, params.embodiedFactor),
-			v1.GCO2eqkWh,
-		),
+	eventInstance.EmbodiedEmissions = v1.NewResourceEmission(
```

Comment:
we should maybe also look at using pointers here with mutex locks instead of doing values

Reply:
That's a really good point. Ideally, there should only be one addition to each field. i.e. embodiedEmissions should only be assigned once. We could have an `if not nill` check before updating and error out for now.